Bug: Cannot access struct member when iterating over array of imported struct types

## Description

When using `for_each` to iterate over an array of structs where the struct type is imported from a library module, the type checker incorrectly reports:

```
error[E4011]: cannot access member 'X' on type 'Y' (not a struct)
```

Even though the type IS a struct.

## Reproduction

**lib/types.ez:**
```ez
module lib

const ItemType enum {
    WEAPON
    ARMOR
    POTION
}

const Item struct {
    name string
    type ItemType
    value u32
    damage u32
}

const Hero struct {
    name string
    inventory [Item]
}
```

**main.ez:**
```ez
module main

import & use @std, @arrays
import lib"./lib"

do main() {
    temp hero = new(lib.Hero)
    hero.name = "TestHero"

    temp sword = lib.Item{
        name: "Sword",
        type: lib.ItemType.WEAPON,
        value: 100,
        damage: 10
    }

    arrays.append(hero.inventory, sword)

    for_each item in hero.inventory {
        println(item.type)  // ERROR: cannot access member 'type' on type 'Item' (not a struct)
    }
}
```

## Expected Behavior

The code should compile and run, printing the enum value of `item.type`.

## Actual Behavior

```
error[E4011]: cannot access member 'type' on type 'Item' (not a struct)
  --> main.ez:22:22
   |
22 |         println(item.type)
   |                      ^ type does not support member access
```

## Observations

- **Works**: Locally defined structs (same file)
- **Works**: Direct struct member access (not in `for_each`)
- **Fails**: Imported struct types accessed within `for_each` loops
- **Strange**: Using the same member in a `when` statement works (e.g., `when item.type { ... }`)

## Environment

- EZ Version: dev (tested against 0.26.0 behavior)
- OS: macOS

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -373,6 +373,7 @@ func (tc *TypeChecker) GetType(name string) (*Type, bool) {
 
 // getStructTypeIncludingModules looks up a struct type by name, checking both local
 // and module types. For qualified names like "lib.Hero", it looks up in moduleTypes.
+// For unqualified names like "Item", it also searches through all registered modules.
 func (tc *TypeChecker) getStructTypeIncludingModules(typeName string) (*Type, bool) {
 	// First check local types
 	if t, exists := tc.types[typeName]; exists && t.Kind == StructType {
@@ -390,6 +391,15 @@ func (tc *TypeChecker) getStructTypeIncludingModules(typeName string) (*Type, bo
 					return t, true
 				}
 			}
+		}
+	}
+
+	// For unqualified names, search through all registered modules
+	// This handles cases where a struct field type like "[Item]" references
+	// a type from the same module without qualification
+	for _, moduleTypes := range tc.moduleTypes {
+		if t, exists := moduleTypes[typeName]; exists && t.Kind == StructType {
+			return t, true
 		}
 	}
 

